frame/dvm/evm/precompiles/bridge/s2s/src/lib.rs

```diff
@@ -159,4 +164,22 @@ where
 		.map_err(|_| custom_precompile_err("encode send message call failed"))?;
 		Ok(abi_encode_bytes(encoded.as_slice()))
 	}
+
+	fn build_message_payload(
+		dvm_parser: &DvmInputParser,
+		caller: H160,
+	) -> Result<Vec<u8>, PrecompileFailure> {
+		let info = S2sMessagePayloadInfo::abi_decode(dvm_parser.input)
+			.map_err(|_| custom_precompile_err("abi_decode message payload info failed"))?;
+
+		let payload = <T as from_substrate_issuing::Config>::ToPangoroMessagePayload {
```

Comment:
```suggestion
		let payload = <T as from_substrate_issuing::Config>::ToPangoroMessagePayload {
```

This is a general precompile used in s2s message delivery. you cannot introduce things like `ToPangoroMessagePayload`. `ToPangoroMessagePayload` is a chain runtime concept.  I suggest refactoring the `encode_send_message_dispatch_call` S2sSendMessageParams to include necessary field for decode payload, like spec_version, weight. @xiaoch05

Reply:
that's right. this method is the generic version of `encode_send_message_dispatch_call`. I will change ToPangoroMessagePayload to generic payload.